File: Stack.js
class Node {
  constructor(data) {
    this.data = data;
    this.next = null;
  }
}

class Stack {
  constructor() {
    this.top = null;
  }

  // Push operation (O(1))
  push(value) {
    const newNode = new Node(value);
    newNode.next = this.top;
    this.top = newNode;
  }

  pop() {
    if (this.top === null) {
      console.log("Stack Underflow");
      return -1;  // Error value
    }
    const popped = this.top.data;
    this.top = this.top.next;
    return popped;
  }

  // Peek operation (O(1))
  peek() {
    if (this.top === null) {
      console.log("Stack is empty");
      return -1;  // Error value
    }
    return this.top.data;
  }

  // Check if stack is empty
  isEmpty() {
    return this.top === null;
  }


}

const stack = new Stack();

stack.push(10);
stack.push(20);
stack.push(30);

console.log("Top element:", stack.peek());  // 30
console.log("Popped:", stack.pop());        // 30
console.log("Top element:", stack.peek());       // 20

File: Stack.py
class Node:
    def __init__(self, data=None):
        self.data = data
        self.next = None

class Stack:
    def __init__(self):
        self.top = None
    
    def push(self, data):
        new_node = Node(data)
        new_node.next = self.top
        self.top = new_node
    
    def pop(self):
        if not self.top:
            return None
        data = self.top.data
        self.top = self.top.next
        return data
    
    def peek(self):
        return self.top.data if self.top else None
    
    def is_empty(self):
        return self.top is None
    
    def display(self):
        current = self.top
        while current:
            print(current.data, end=" -> ")
            current = current.next
        print("None")

if __name__ == "__main__":
    stack = Stack() # stack object
    stack.push(1)
    stack.push(2)
    stack.push(3)
    stack.display()
    print("Peek:", stack.peek())
    print("Pop:", stack.pop())
    stack.display()